classes/Hook/HookDisplayBackOfficeHeader.php

```diff
@@ -48,84 +51,139 @@ public function __construct(Ps_Googleanalytics $module, Context $context)
-                                    'purchase',
-                                    $eventData,
-                                    ['event_callback']
-                                );
-                            }
-                        }
-                    }
-                }
-            }
+        $gaScripts = '';
+
+        // Render base tag using displayHeader hook with backoffice parameter
+        $gaScripts .= $this->module->hookDisplayHeader(null, true);
+
+        // Process manual orders instantly, we have their IDs in cookie
+        $gaScripts .= $this->processManualOrders();
+
+        // Backload old orders that failed to load normally
+        $gaScripts .= $this->backloadFailedOrders();
+
+        return $gaScripts;
+    }
+
+    /**
+     * Checks if there are any orders that failed to be sent normally through front office and processes them
+     */
+    protected function backloadFailedOrders()
+    {
+        $gaScripts = '';
+        if (empty(Configuration::get('GA_BACKLOAD_ENABLED'))) {
+            return $gaScripts;
+        }
+
+        // Check for value on how long back we will get them
+        $backloadDays = (int) Configuration::get('GA_BACKLOAD_DAYS');
+        if ($backloadDays < 1) {
+            return $gaScripts;
+        }
+
+        // Get all failed orders (either not present in our table or not sent)
+        // We go GA_BACKLOAD_DAYS into the past and at least 30 minutes old
+        $failedOrders = Db::getInstance()->ExecuteS(
+            'SELECT DISTINCT o.id_order, g.sent FROM `' . _DB_PREFIX_ . 'orders` o
+            LEFT JOIN `' . _DB_PREFIX_ . GanalyticsRepository::TABLE_NAME . '` g ON o.id_order = g.id_order
+            WHERE (g.sent IS NULL OR g.sent = 0) AND o.date_add BETWEEN NOW() - INTERVAL ' . $backloadDays . ' DAY AND NOW() - INTERVAL 30 MINUTE'
+        );
+
+        // Process each failed order
+        foreach ($failedOrders as $row) {
+            $gaScripts .= $this->processOrder((int) $row['id_order']);
+        }
+
+        return $gaScripts;
+    }
+
+    /**
+     * Checks if there are any manual orders in cookie and processes them
+     */
+    protected function processManualOrders()
+    {
+        $gaScripts = '';
+        $adminOrders = $this->context->cookie->__get('ga_admin_order');
+        if (empty($adminOrders)) {
+            return $gaScripts;
+        }
+
+        // Separate them by IDs and process one by one
+        $adminOrders = explode(',', $adminOrders);
+        foreach ($adminOrders as $idOrder) {
+            $gaScripts .= $this->processOrder((int) $idOrder);
+        }
+
+        // Clean up the cookie
+        $this->context->cookie->__unset('ga_admin_order');
+        $this->context->cookie->write();
+
+        return $gaScripts;
+    }
+
+    /**
+     * Renders tracking code for given order
+     *
+     * @param int $idOrder
+     */
+    public function processOrder($idOrder)
```

Comment:
Same here ;) PHPDoc and descriptive function name.

In an ideal world we should be able to guess the function content just by reading the name